src/nupic/algorithms/Connections.hpp

```diff
@@ -425,7 +425,8 @@ class Connections : public Serializable
    * @param increment  Change in permanence for synapses with active presynapses.
    * @param decrement  Change in permanence for synapses with inactive presynapses.
    */
-  void adaptSegment(const Segment segment, SDR &inputs,
+  void adaptSegment(const Segment segment, 
+		    const SDR &inputs,
```

Comment:
@ctrl-z-9000-times do you use adapt segment in CP? Otherwise it's only used in SP, so I'd like to move it there.

Reply:
I think this belongs here.  For code organization, I want all of the generic tools for synapses & segments in the Connections class.  
* The CP will probably use this method, after it's written.
* The TM does not use this method but with some simple modifications could use this method.

Reply:
sounds fair. 

> The TM does not use this method but with some simple modifications could use this method. 

might be nice if TM used Segments this way as well. For improved biological detail as well as shared codepaths.